client/packages/common/src/ui/layout/tables/material-react-table/useMaterialTableColumns.tsx

```diff
@@ -7,95 +7,100 @@
-        const filterKey = (mrtProperties.accessorKey ||
-          mrtProperties.id) as string;
-
-        const columnDefProperties: Partial<MRT_ColumnDef<T>> = {};
-
-        // Add canonical filter definitions
-        switch (filterType) {
-          case 'none':
-            columnDefProperties.enableColumnFilter = false;
-            break;
-          case 'text':
-            filterUpdaters[filterKey] = (value: string) => {
-              updateQuery({ [filterKey]: value });
-            };
-            break;
-
-          case 'enum':
-            columnDefProperties.filterVariant = 'select';
-            columnDefProperties.filterSelectOptions = filterValues;
-            filterUpdaters[filterKey] = (value: string) => {
-              updateQuery({ [filterKey]: value });
-            };
-            break;
-          // case 'number':
-          //   TO-DO
-          case 'dateRange':
-            columnDefProperties.filterVariant = 'date-range';
-            filterUpdaters[filterKey] = ([date1, date2]: [
-              Date | '',
-              Date | '',
-            ]) => {
-              updateQuery({
-                [filterKey]: {
-                  from: date1 ? customDate(date1, urlQueryDateTime) : '',
-                  to: date2 ? customDate(date2, urlQueryDateTime) : '',
-                },
-              });
-            };
-            break;
-        }
-
-        // Add alignment styling
-        if (align) {
-          columnDefProperties.muiTableBodyCellProps = params =>
-            mergeCellProps(
+  const simplifiedMobileView = useSimplifiedTabletUI();
+  const tableDefinition = useMemo(() => {
+    const columns = omsColumns
+      .filter(col => col.includeColumn !== false)
+      .map(col => {
+        if (col.align) {
+          col.muiTableBodyCellProps = params => {
+            // Add alignment styling
+            return mergeCellProps(
               {
                 sx:
-                  align === 'right'
+                  col.align === 'right'
                     ? {
                         justifyContent: 'flex-end',
                         paddingRight: '2em', // Padding to account for header icons
                       }
-                    : align === 'center'
-                      ? {
-                          justifyContent: 'center',
-                          // To-DO: Add padding for center aligned cells
-                        }
+                    : col.align === 'center'
+                      ? // To-DO: Add padding for center aligned cells
+                        { justifyContent: 'center' }
                       : {},
               },
               params
             );
+          };
         }
+        return col;
+      });
+
+    const defaultHiddenColumns = simplifiedMobileView
+      ? columns
+          .filter(col => col.defaultHideOnMobile)
+          .map(col => String(col.id ?? col.accessorKey))
```

Comment:
These will be passed as the default hidden columns, if no pref is stored in localstorage for this table yet